cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,567 @@
+from mpl_toolkits.mplot3d import Axes3D
+
+
+class RabiExperimentResult(object):
+    """Class for storing and plotting data from a Rabi oscillation
+    experiment.
+    """
+
+    def __init__(self, num_shots: int, rabi_angles: Sequence[float],
+                 excited_state_pops: Sequence[float]):
+        """
+        Args:
+            num_shots: The number of repetitions of the circuit for each Rabi
+              angle.
+            rabi_angles: The rotation angles of the qubit around the x-axis
+              of the Bloch sphere.
+            excited_state_pops: The corresponding probabilities that the
+            qubit is in the excited state.
+        """
+        self._num_shots = num_shots
+        self._rabi_angles = rabi_angles
+        self._excited_state_pops = excited_state_pops
+
+    @property
+    def data(self) -> Dict[str, Sequence[float]]:
+        return {'rabi_angles': self._rabi_angles,
+                'excited_state_probability': self._excited_state_pops}
+
+    def plot(self) -> None:
+        fig = pyplot.figure()
+        pyplot.plot(self._rabi_angles, self._excited_state_pops, 'ro-',
+                    figure=fig)
+        pyplot.xlabel(r"Rabi Angle ($\pi$)")
+        pyplot.ylabel('Excited State Probability')
+
+
+class RandomizedBenchmarkingResult(object):
+    """Class for storing and plotting data from a randomized benchmarking
+    experiment.
+    """
+
+    def __init__(self, num_cfds_seq: Sequence[float],
+                 gnd_state_pops: Sequence[float]):
+        """
+        Args:
+            num_cfds_seq: The different numbers of Cliffords in the RB study.
+            gnd_state_pops: The corresponding average ground state
+              probabilities.
+        """
+        self._num_cfds_seq = num_cfds_seq
+        self._gnd_state_pops = gnd_state_pops
+
+    @property
+    def data(self) -> Dict[str, Sequence[float]]:
+        return {'number_of_cliffords': self._num_cfds_seq,
+                'ground_state_probability': self._gnd_state_pops}
+
+    def plot(self) -> None:
+        fig = pyplot.figure()
+        pyplot.plot(self._num_cfds_seq, self._gnd_state_pops, 'ro-', figure=fig)
+        pyplot.xlabel(r"Number of Cliffords")
+        pyplot.ylabel('Ground State Probability')
+
+
+class StateTomographyResult(object):
+    """Class for storing and plotting a density matrix obtained from a state
+    tomography experiment."""
+
+    def __init__(self, density_matrix: numpy.ndarray):
+        """
+        Args:
+            density_matrix: The density matrix obtained from tomography.
+        """
+        self._density_matrix = density_matrix
+
+    @property
+    def data(self) -> numpy.ndarray:
+        return self._density_matrix
+
+    def plot(self) -> None:
+        _plot_density_matrix(self._density_matrix)
+
+
+def rabi_oscillations(sampler: sim.SimulatesSamples, qubit: devices.GridQubit,
+                      final_angle: float, num_shots: int, num_points: int) \
```

Comment:
Nit: don't use `\` for line continuation. Put the close parens on a new line instead.

Reply:
Done. \ is no longer used for line continuations anywhere.